### PR TITLE
Create main.yml for CI

### PR DESCRIPTION
**Pull request by soaptherope**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Cache Maven packages
+      uses: actions/cache@v3  
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    - name: Build and Test
+      run: mvn clean test
```
Comment (Alexmansar) [inline .github/workflows/main.yml]: image
Hi) just update your solution to build jar files)

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@soaptherope 
Good practice  👍 
please, check 1 comment

File: .github/workflows/main.yml (hunk 1 above)
Comment (soaptherope) [reply .github/workflows/main.yml]: is it just adding "package" at the end? so it packages application into jar in target folder?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Cache Maven packages
+      uses: actions/cache@v3  
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    - name: Build and Test and Package JAR
+      run: mvn clean test package
```
Comment (Alexmansar) [inline .github/workflows/main.yml]: could be just clean package
but, also you should provide more locic
try
```
- name: Build with Maven
      run: mvn -B package --file pom.xml

    - name: Upload JAR Artifact
      uses: actions/upload-artifact@v3
      with:
        name: your_app_name
        path: target/*.jar
```
as a result it shoulb be a jar file

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@soaptherope try to fix the comment please)

File: .github/workflows/main.yml (hunk 2 above)
Comment (soaptherope) [reply .github/workflows/main.yml]: got it, thanks!

**Review by Alexmansar — APPROVED** [review]
@soaptherope Could be merged as well))

image

can you download it and run in console? I can't install new version on my laptop to check it)

**Comment by Alexmansar** [comment]
@soaptherope 
image
Ypu should update pom.xml file to fix it

**Pull request merged**